src/gtkspellcheck/spellcheck.py

```diff
@@ -329,7 +345,12 @@ def recheck(self):
         Rechecks the spelling of the whole text.
         """
         start, end = self._buffer.get_bounds()
-        self.check_range(start, end, True)
+
+        if self._batched_rechecking and end.get_offset() > _BATCHING_THRESHOLD_CHARS:
+            start_mark = self._buffer.create_mark(None, start)
+            self._continue_batched_recheck(start_mark)
+        else:
+            end = self.check_range(start, end, True)
```

Comment:
Why is the result of `check_range` assigned to `end`?

Otherwise, this looks good now! 👍

Reply:
Oops, that's because I was a hack (in a rush) :facepalm: 

(FWIW it's a weird out take from a version of the change I had where `check_range` returned its more accurate end iter as the starting off point for the next batch... but as the difference in the possible single word being double checked wasn't worth the code mess I culled it. Not sure why it ended up on that line though, some odd copy/paste.)